src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -316,6 +316,28 @@ Copyright (c) .NET Foundation. All rights reserved.
   
   <UsingTask TaskName="ResolveRuntimePackAssets" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
+  <!--
+    ============================================================
+                                        IncludeTargetingPackReference
+
+    Add package reference for required targeting pack if necessary
+    ============================================================
+    -->
+  <Target Name="IncludeTargetingPackReference" BeforeTargets="_GetRestoreSettingsPerFramework" Condition="'$(TargetFrameworkMoniker)' != '' and '$(TargetFrameworkIdentifier)' == '.NETFramework'">
```

Comment:
I think we should probably also have this run before _CheckForInvalidConfigurationAndPlatform so the `PackageReference` item gets added during the normal build as well as for restore.  I don't think there's any hard requirement for this, but it feels more consistent.

Let's also put the target behind a condition based on an `AutomaticallyUseReferenceAssemblyPackages` property which defaults to true, in case for some reason someone wants to opt out of this.

Reply:
Escape hatch opt out is a great idea.

Reply:
Added in the latest update: https://github.com/dotnet/sdk/pull/10981/commits/fb3772a07b7ebd2d41606cae85863e8d920b2d34#diff-81720bb5fb8b13c6dff8490812b4c44aR327